nativelib/src/main/scala/scala/scalanative/unsigned/UByte.scala

```diff
@@ -263,6 +263,13 @@ final class UByte private[scalanative] (
 
   @inline final override def toString(): String = toInt.toString()
 
+  @inline override def hashCode(): Int = underlying.##
+
+  @inline override def equals(obj: Any): Boolean = obj match {
+    case that: UByte => this == that
```

Comment:
We should compare the `underlying`s, otherwise we'll have to go again through the dispatch of `__scala_==`, although we already know that we have a specific type:
```suggestion
    case that: UByte => this.underlying == that.underlying
```
The same applies for all the other types, of course.